sklearn/neighbors/_base.py

```diff
@@ -746,10 +747,18 @@ class from an array representing our data set and ask who's
 
         n_samples_fit = self.n_samples_fit_
         if n_neighbors > n_samples_fit:
-            raise ValueError(
-                "Expected n_neighbors <= n_samples, "
-                " but n_samples = %d, n_neighbors = %d" % (n_samples_fit, n_neighbors)
-            )
+            if X is None:
+                raise ValueError(
+                    "Expected n_neighbors < n_samples, "
+                    " but n_samples = %d, n_neighbors = %d" %
+                    (n_samples_fit, original_n_neighbors)
+                )
+            else:
+                raise ValueError(
+                    "Expected n_neighbors <= n_samples, "
+                    " but n_samples = %d, n_neighbors = %d" %
+                    (n_samples_fit, original_n_neighbors)
+                )
```

Comment:
I think we can reduce some repeated code as follows:

```suggestion
            if query_is_train:
                n_neighbors -= 1
                inequality_str = "n_neighbors < n_samples"
            else:
                inequality_str = "n_neighbors <= n_samples"
                
            raise ValueError(
                f"Expected {inequality_str}, but "
                f"n_neighbors = {n_neighbors}, n_samples = {n_samples}"
            )
```

Since we are always raising an error message, I think it's okay to do `n_neighbors -= 1` to adjust the neighbors back up. Also using `query_is_train` makes this more consistent with the code above.